### PR TITLE
Fixed path extraction for Windows and OSX 

### PR DESCRIPTION
**Pull request by bwklein**
Corrects regex for file:\\ pattern from #12 and adds in a pattern for matching quoted path string.

### CODE REVIEW

**Comment by ldez** [comment]
your PR breaks the package :smile:

**Comment by bwklein** [comment]
Need to update test spec for new code.

**Comment by ldez** [comment]
wait I report your fix

**Comment by ldez** [comment]
I going to switch on windows to test.

**Comment by bwklein** [comment]
@ldez Can you check if I did it right in the updated spec?

**Comment by bwklein** [comment]
It looks like the tests are working as expected now.

**Comment by ldez** [comment]
That doesn't work on Win7... You are using Win10?

**Comment by ldez** [comment]
why you have removed `^` and `$` ?

**Comment by bwklein** [comment]
I am testing on Windows 10 and OS X now.  It works for both systems for me.  What string is Windows 7 using for the path?

`^` and `$` could probably be added back in.  It wasn't working for me when I left those in at first.  I think because for my file:\\ URL there is not a double-quote at the end of the string and there was no need to indicate anything at the end of it.  I have added both symbols back into the double-quote stripping pattern.

**Comment by ldez** [comment]
the `$` is use when you try to copy multiple file:

```
file:\\c:\img01.png
file:\\c:\img02.png
```

**Comment by ldez** [comment]
currently, we not support this behavior

File: lib/main.coffee [hunk 1]
```diff
@@ -85,7 +85,17 @@ module.exports =
...
+          console.log clipboardText
+
+          # windows specific
+          windowsFilePattern = /^file:[\/]{2,3}(.*)/
+          windowsPathPattern = /^\"(.*)\"$/
+          if clipboardText.match windowsFilePattern
+            clipboardText = windowsFilePattern.exec(clipboardText)[1]
+
+          if clipboardText.match windowsPathPattern
+            clipboardText = windowsPathPattern.exec(clipboardText)[1]
```
Comment (ldez) [inline lib/main.coffee]: can you rewrite this part like that:

``` coffee

# windows specific
windowsFilePattern = /^file:[\/]{2,3}(.*)$/
windowsPathPattern = /^\"(.*)\"$/
if clipboardText.match windowsFilePattern or clipboardText.match windowsPathPattern
  clipboardText = windowsPathPattern.exec(clipboardText)?[1] or windowsFilePattern.exec(clipboardText)?[1]
```

File: spec/main-url-support-spec.coffee [hunk 2]
```diff
@@ -33,7 +33,7 @@ describe 'URL with AsciiDoc Image helper should', ->
     temp.cleanupSync()
 
 
-  it 'create a link and store the image in a directory when image folder append to link and URL in "windows style" and use the default directory', ->
+  it 'create a link and store the image in a directory when image folder append to link and URL in "file" format and use the default directory', ->
```
Comment (ldez) [inline spec/main-url-support-spec.coffee]: "windows file" instead of "file"

File: spec/main-url-support-spec.coffee [hunk 3]
```diff
@@ -47,11 +47,40 @@ describe 'URL with AsciiDoc Image helper should', ->
     expect(editor.getSelectedText()).toMatch /^foobar$/
     editor.delete()
 
-    imageUrl = '"file:///' + path.join(__dirname, 'fixtures', imageName) + '"'
+    imageUrl = 'file:///' + path.join(__dirname, 'fixtures', imageName) + ''
```
Comment (ldez) [inline spec/main-url-support-spec.coffee]: remove `+ ''`

**Comment by ldez** [comment]
On Win7 the feature doesn't work because the string is always empty.

File: lib/main.coffee [hunk 4]
```diff
@@ -85,7 +85,17 @@ module.exports =
 
         # Image URL support
         else if atom.config.get 'asciidoc-image-helper.enableUrlSupport'
-          clipboardText = clipboard.readText().split(/file:[\/]{2,3}/).join('').replace /^\"|\"$/g, ''
+          clipboardText = clipboard.readText()
+          console.log clipboardText
```
Comment (ldez) [inline lib/main.coffee]: can you remove `console.log` ?

**Pull request merged**